### ISSUE TITLE
Require default export on `<script>`

### ISSUE THREAD

**Issue by privatenumber** [vuejs/eslint-plugin-vue#2383]
**Please describe what the rule should do:**

To warn when there's a `<script>` tag with no default export.

With the new `<script setup>` syntax, I've noticed cases where devs forget or miss the `setup` and can't figure why their code is not working correctly.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- Forgetting to add `setup` -->
<script>
const value = 1;
</script>
```

**Additional context**

From: https://github.com/vuejs/eslint-plugin-vue/issues/2328

**Comment by ota-meshi** [comment]
Thank you for the rule suggestion! Sounds good to me.
